Feature/select run

nmenezes0 opened the pull request:
## Context

<!-- Why are you making this change? What might surprise someone about it? -->
Allow users to select the correct processing run.

## Changes proposed in this pull request

<!-- If there are UI changes, please include Before and After screenshots. -->
No change if no themes for a given consultation:
![image](https://github.com/user-attachments/assets/b2a1dfb1-2458-4c43-be51-140271e0d1e4)

Added a drop-down to select different processing runs:
![image](https://github.com/user-attachments/assets/293dd7d6-8f58-4604-b479-56cd3f22cb0d)

## Guidance to review

<!-- How could someone else check this work? Which parts do you want more feedback on? -->
Frontend help required please!

- How should I include this drop-down? Don't know how to format it?
- Is this even the right page to put it on? (I think so, then we can explore questions/responses for that particular run - and the specific changes are likely to change a lot anyway.)
- Is there a clearer way to explain the processing runs? I've used "Themes generated at" - but might be a better way to explain.

## Link to JIRA ticket

<!-- e.g. https://technologyprogramme.atlassian.net/jira/software/c/projects/ER/boards/346?selectedIssue=ER-87 -->
Finally finishes: https://technologyprogramme.atlassian.net/jira/software/projects/CON/boards/418?selectedIssue=CON-370

## Things to check

- [x] I have added any new ENV vars in all deployed environments and updated the `.env.example` and `.env.test` files in the repo - N/A

KevinEtchells commented on consultation_analyser/consultations/jinja2/consultations/consultations/show.html:
```diff
@@ -18,6 +18,32 @@ <h1 class="govuk-heading-m">{{ page_title }}</h1>
+      <div class="govuk-grid-column-full">
+          <div class="iai-panel">
+            <form>
+              <div class="govuk-form-group">
```
```suggestion
              <div class="govuk-form-group govuk-!-margin-bottom-3">
```

nmenezes0 replied:
Thanks - have added that suggestion. Do you think there should be a gap here?
![image](https://github.com/user-attachments/assets/25feb74a-5f0d-45b4-98f8-df65d8a4d704)

nmenezes0 replied:
Scrap that - am removing from that page anyway!

KevinEtchells replied:
Just noticed that they're two separate panels. In which case, yes. And would be good to get the left-alignment matching. I'll do an update for that.

nmenezes0 replied:
Have removed from this page (actually doesn't really make sense to have a dropdown here) - but good to know!

KevinEtchells reviewed (approved):
Looks good to me. Just one tiny suggestion added (to reduce the gap between the select box and the update button), but feel free to ignore!

nmenezes0 commented:
Duncan's comments from Slack:

- should the UI to pick a theme run be present on the overall consultation page or on the question pages (or both!?)
- related, there's nothing on the question pages to indicate which run we're looking at
- if there is only one run to choose from, could we skip showing the dropdown and just show text?

nmenezes0 commented:
> Duncan's comments from Slack:
> 
> * should the UI to pick a theme run be present on the overall consultation page or on the question pages (or both!?)

I now think it's confusing to have it on the consultation page - because you can't really tell the underlying themes. I now have the dropdown. it on the question page and the responses page.

> * related, there's nothing on the question pages to indicate which run we're looking at

Now changed as we have the drop-down on that page.

> * if there is only one run to choose from, could we skip showing the dropdown and just show text?

Yes - that change now in.

nmenezes0 commented:
Updated question page:
![image](https://github.com/user-attachments/assets/02ea58ed-62c4-4afa-b274-939903c7909e)

Updated response page:
![image](https://github.com/user-attachments/assets/1f8ac518-e3b2-4995-bcf8-232aa121dc7a)

If there's just one processing run:
![image](https://github.com/user-attachments/assets/1a802b53-3176-4655-acf7-b218bccc90e5)

No dropdown shown if there are no themes.

When a different run is selected and updated using the dropdown, the filters are cleared (as e.g. themes might change, so it doesn't make sense to preserve them).

nmenezes0 commented:
Closing as this is no longer relevant.
